orquesta/conducting.py

```diff
@@ -882,3 +888,59 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_workflow_rerun(self, options):
+        # Check current workflow status.
+        if self.get_workflow_status() not in statuses.ABENDED_STATUSES:
+            raise exc.InvalidWorkflowRerunStatus()
+
```

Comment:
So, I think instead of processing one task at a time and then raising an error when a task is not found or status doesn't checks out.  It's probably better to do that check up front before you set the workflow execution status to RESUMING. 

```
failed_rerun_tasks = []
rerun_tasks = []

# Identify tasks that are either invalid or cannot be rerun and
# raise an exception before rerunning the workflow execution.
for task_id in options.get('tasks', []):
    try:
        task_route = 0
        task = self.workflow_state.get_task(task_id, task_route)
    except Exception:
        failed_rerun_tasks.append(task_id)
        continue

    if task['status'] not in statuses.ABENDED_STATUSES:
        failed_rerun_tasks.append(task_id)
        continue

    rerun_tasks.append(task)

if failed_rerun_tasks:
    raise exc.InvalidRerunTasks(failed_rerun_tasks)

# Force reset workflow status to running
self.workflow_state.status = statuses.RESUMING

# Process the tasks that are identified for rerun.
for task in rerun_tasks:
    ctxs = copy.deepcopy(task['ctxs']['in'])
    route = copy.deepcopy(task['route'])
    prev = copy.deepcopy(task['prev'])

    # Add a new task entry
    self.add_task_state(
        task_id,
        route,
        in_ctx_idxs=ctxs,
        prev=prev
    )
	
    # Create a new entry in staging for a failed task.
    self._workflow_state.add_staged_task(
        task_id,
        route,
        ctxs=ctxs,
        ready=True
    )
	
    self.reset_errors_for_rerun_task(task_id)

self.reset_workflow_output()
```

The InvalidRerunTasks exception should have the following message format.
```
Returns the message "Unable to rerun workflow because one or more tasks is not found or not in failed status: %s" % ', '.join(failed_rerun_tasks)
```